workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/CodeGeneration/Model/SchemaBundleV1/TypeDefinitionRaw.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Improbable.Gdk.CodeGeneration.Model.SchemaBundleV1
+{
+    public class TypeDefinitionRaw
+    {
+        [JsonProperty("identifier")] public Identifier Identifier;
+        [JsonProperty("fieldDefinitions")] public List<Field> Fields;
+
+        public class Field
+        {
+            [JsonProperty("identifier")] public Identifier Identifier;
+            [JsonProperty("fieldId")] public uint FieldId;
+            [JsonProperty("transient")] public bool IsTransient;
+            [JsonProperty("listType")] public ListType List;
+            [JsonProperty("mapType")] public MapType Map;
+            [JsonProperty("optionType")] public OptionType Option;
+            [JsonProperty("singularType")] public SingularType Singular;
+
+            public class ListType
+            {
+                [JsonProperty("innerType")] public InnerType InnerType;
+            }
+
+            public class MapType
+            {
+                [JsonProperty("keyType")] public InnerType KeyType;
+                [JsonProperty("valueType")] public InnerType ValueType;
+            }
+
+            public class OptionType
+            {
+                [JsonProperty("innerType")] public InnerType InnerType;
+            }
+
+            public class SingularType
+            {
+                [JsonProperty("type")] public InnerType Type;
+            }
+
+            // Note: Mutually exclusive
+            public class InnerType
+            {
+                [JsonProperty("type")] public UserType UserType;
```

Comment:
Is this exclusivity verified anywhere?

Reply:
Nope, but there is this: https://www.newtonsoft.com/jsonschema which you can validate the structure of the schema. It looks like you can use the structure defined here to generate the validation schema